hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/SqmPluralPartJoin.java

```diff
@@ -111,4 +114,15 @@ public String toString() {
 				getReferencedPathSource().getPathName()
 		);
 	}
+
+	@Override
+	public DomainResult<T> createDomainResult(
+			String resultVariable, DomainResultCreationState creationState) {
+		throw new NotYetImplementedFor6Exception( getClass() );
+	}
+
+	@Override
+	public void applySqlSelections(DomainResultCreationState creationState) {
+		throw new NotYetImplementedFor6Exception( getClass() );
+	}
```

Comment:
Just like you did for `SqmSingularJoin`, `SqmPluralPartJoin`, `SqmRoot`, `SqmDynamicInstantiation`, `SqmFunction`, `SqmLiteral`, `SqmCaseSearched`, `SqmJpaCompoundSelection` and `SqmCollectionSize` don't need to implement `DomainResultProducer` because these are always translated to `SqmPathInterpretation`, so you can remove these dummy implementations.